src/analyze_includes/test/parse_source_test.py

```diff
@@ -72,12 +72,12 @@ def test_filter_includes_for_patterns(self):
                 Include(file=Path("file3"), include="foo/bar.h"),
                 Include(file=Path("file4"), include="foo/nested/bar.h"),
                 Include(file=Path("file4"), include="baz_some_partial_name.h"),
-                Include(file=Path("file5"), include="match_substring"),
+                Include(file=Path("file5"), include="match_substring_not_sufficient"),
             ],
             ignored_includes=IgnoredIncludes(paths=[], patterns=["foo/.*", ".*some_partial_name.h", "substring"]),
         )
 
-        self.assertEqual(len(result), 2)
+        self.assertEqual(len(result), 3)
```

Comment:
I am also fine with this counter proposal to show explicitly substring matching is not happening without regex pattern. However, in this case also make sure also add `self.assertTrue(Include(file=Path("file5"), include="match_substring_not_sufficient") in result)` below to make sure all 3 elements of the result are as expected

Reply:
Aah, I did't realize what you mean by `change the test case as requested` at the first sight. Now I got it. Done.